src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -333,67 +366,85 @@ pal::string_t fx_muxer_t::resolve_fx_dir(host_mode_t mode, const pal::string_t&
         return pal::string_t();
     }
 
-    auto fx_dir = own_dir;
-    append_path(&fx_dir, _X("shared"));
-    append_path(&fx_dir, fx_name.c_str());
+    // Multi-level SharedFX lookup will look for the most appropriate version in several locations
+    // by following the priority rank below (from 1 to 4):
+    // 1. Current working directory
+    // 2. User directory
+    // 3. .exe directory
+    // 4. Global .NET directory
+    // If it is not activated, then only .exe directory will be considered
+
+    std::vector<pal::string_t> hive_dir;
+    pal::string_t env_lookup;
+
+    // Multi-level SharedFX lookup happens only if DOTNET_MULTILEVEL_LOOKUP env var is set to 1
+    bool multilevel_lookup = pal::getenv(_X("DOTNET_MULTILEVEL_LOOKUP"), &env_lookup) && (env_lookup == _X("1"));
```

Comment:
As discussed the other day, please change this to be enabled by default and use the env var to be used for opt-out to ensure we are testing the repo with the feature enabled.